ext/symengine/ruby_real_mpfr.c

```diff
@@ -1,7 +1,22 @@
 #include "ruby_real_mpfr.h"
 
-VALUE crealmpfr_init(VALUE self, VALUE num_value) {
-    
-    return self;
+VALUE crealmpfr_init(VALUE self, VALUE num_value, VALUE prec_value) {
+    basic_struct *cresult;
+    VALUE result;
+    double d;
+    int prec = NUM2INT(prec_value);
+    cresult = basic_new_heap();
+
+    switch( TYPE(num_value) ) {
+        case T_FLOAT:
+            d = RFLOAT_VALUE(num_value);     
+            real_mpfr_set_d(cresult, d, prec);
+            break;
+        default:
+            rb_raise(rb_eTypeError, "Invalid Type: Float, BigDecimal or String required.");
+            break;
+    }
+    result = Data_Wrap_Struct(c_real_mpfr, NULL, cbasic_free_heap, cresult);
+    return result;
```

Comment:
I'm getting a `terminate called after throwing an instance of 'Teuchos::NullReferenceError'` error right after the return of this code. I've been trying to find out with no luck.

The real_mpfr_set_d function is working, it's being tested in [here] (https://github.com/symengine/symengine/pull/972/files#diff-8491b7e47a679403840abc38c4e23cf4R200)

Also, I used print statements to make sure that the code runs up until the return statement.

The error is given when I give a sample input like this:
```
A = SymEngine::RealMPFR.new(10.4, 10)
```

@abinashmeher999 @isuruf